controllers/cloud.redhat.com/clowdapp_controller.go

```diff
@@ -380,9 +383,14 @@ func (r *ClowdAppReconciler) SetupWithManager(mgr ctrl.Manager) error {
 			&source.Kind{Type: &crd.ClowdEnvironment{}},
 			handler.EnqueueRequestsFromMapFunc(r.appsToEnqueueUponEnvUpdate),
 		).
+		Watches(
+			&source.Kind{Type: &batchv1.Job{}},
+			handler.EnqueueRequestsFromMapFunc(r.appToEnqueueOnJobCompletion),
+		).
 		Owns(&apps.Deployment{}).
 		Owns(&core.Service{}).
 		Owns(&core.ConfigMap{}).
+		// Owns(&batchv1.Job{}).
```

Comment:
Are you using the Watch because the Own can't have a filter enqueue on it? If so, I think the logic we have in the enqueuer is like this because of the attributes setup. In my mind rather than watching all Jobs, which could increased overall memory consumption anyway, watching the ones we own and always triggering this apps reconciliation seems like the good way to go. If we are worried about too many reconciliations because the job status changes too often, we have just redone the predicate system to make it FAR easier to define a customer predicate filter to only allow certain changes to the resource to trigger a reconciliation.